### DISCUSSION TITLE
Strategie to use one CSS file for multiple HTML files

### DISCUSSION THREAD

**Discussion by buhtz** [pygments/pygments#2387] [Q&A] [answered]
I have round about 150 code blocks in 100 HTML files. There are different languages: Python, R, bash, unknown, HTML, ... Some of the HTML files do contain multiple code blocks with different languages.

I read some of the docs and would say it is the most efficient to use one CSS-file for all of the 100 HTML files.

How can I realize that. The pygments css classes (e.g. `p`, `nb`, ...) are specific to the used language (python, cpp, ...) or not? I think that is the key question.

I need to run `highlight()` for each code block (150 times). When using `full=true` I could extract the related CSS from the `<style>` tag. But I have to join this extractions from each of the 100 files.

I can do `HtmlFormatter().get_style_defs()`. But it seems to me the result is not complete. For example the class `p` is missing in that output. But the `p` is used in python and c++ output.

So I'm confused in which direction I should go here.

Here is an MWE with Python and C++ code blocks

```
from pygments import highlight
from pygments.lexers import get_lexer_by_name
from pygments.formatters import HtmlFormatter

code_python = '''if __name__ == "__main__":
    print("Hello World!")
'''

code_cpp = '''#include <iostream>
int main()
{
  std::cout << "Hello, World!" << std::endl;
  return 0;
}'''

result = highlight(
    code=code_python,
    lexer=get_lexer_by_name('python'),
    formatter=HtmlFormatter()
)
print(result)

result = highlight(
    code=code_cpp,
    lexer=get_lexer_by_name('python'),
    formatter=HtmlFormatter()
)
print(result)

```

**Comment by birkenfeld** [comment, marked as answer]
The CSS definitions are not dependent on the lexer. "p" is likely simply unstyled in your chosen style (it's short for punctuation, which is commonly left unstyled).

**Reply by buhtz** [reply]
This sounds it isn't that complex than I thought, right?
I just net to take the output of `HtmlFormatter().get_style_defs()` store it into an css file and thats it.
Correct?

**Reply by birkenfeld** [reply]
Yes, that should be sufficient.
